### DISCUSSION TITLE
Dynamic Power limiter erzeugt Schwingungen / und wie Huawei Charger steuern?

### DISCUSSION THREAD

**Discussion by dragricola** [hoylabs/OpenDTU-OnBattery#546] [Q&A] [answered]
Ich bin gerade dabei OpenDTUonBattery in Betrieb zu setzen und habe dazu noch einige Fragen. 

Meine Konfiguration: Batterie: WULILLS 48v 120ah Lifepo4 Battery (bisher ohne aktivierte Schnittstelle via CAN-Bus) , Hoymiles HM-1200, AC-Lader: Huawei R4850G2, Stromzählertyp: HTTP(S) + JSON (Hichi mit Tasmota), MQTT-Server auf iobroker.

Meine Fragen:
1. Die Regeleingriffe des  Dynamic Power limiters führen zu sehr starkem Übersteuern, sodass die Netzleistung sehr stark schwankt und somit auch zeitweise Energie aus der Batterie zurück ins Netz fließt. Wie kann man die Reglerparameter beeinflussen, sodass der Regler stabil läuft? 
Meine Dynamic Power limiter Konfiguration  lautet: Aktiviert, Ausführliche Protokollierung, Aktiviere Solar-Passthrough, Strategie zur Batterieentleerung: Wechselrichter ID: ID 00, Kanal ID: CH0, Angestrebter Netzbezug: 20 W, Hysterese für das berechnete Limit: 10 W,
Unteres Leistungslimit: 10 W, Oberes Leistungslimit: 1200 W.

2. Wie kann OpenDTUonBattery den Huawei AC-Lader steuern oder muss die Steuerung extern über den ioBroker erfolgen? 

3. Die CAN-Schnittstelle der WULILLS 48v 120ah Lifepo4 Battery ist angeblich kompatibel zur Pylontech. Leider konnte bisher noch keine Kommunikation aufgebaut werden. Weiß jemand Rat?

**Comment by schlimmchen** [comment]
> Die Regeleingriffe des Dynamic Power limiters führen zu sehr starkem Übersteuern, sodass die Netzleistung sehr stark schwankt

Dein Verbrauch schwankt stark und unvorhersehbar, der DPL versucht verzweifelt dagegenzuhalten.

> und somit auch zeitweise Energie aus der Batterie zurück ins Netz fließt.

Korrekt. Es gibt immer Szenarien, in denen das passieren wird.

> Wie kann man die Reglerparameter beeinflussen, sodass der Regler stabil läuft?

Welche Regelparameter? Welcher Regler? Der DPL berechnet ein Leistungslimit für den Inverter anhand des Verbrauchs und schickt das an den Inverter. Mehr passiert da nicht.

**Reply by dragricola** [reply]
Vielen Dank für die Antwort zu meiner 1. Frage. 
Wie wird das Leistungslimit im DPL denn berechnet? 
Nach meinem Verständnis sollte der DPL den Huawei so limitieren, dass die Regelabweichung, die sich aus der Abweichung vom angestrebten Netzbezug (Soll-Netzleistung) und der Ist-Netzleistung berechnet, möglichst klein und bei Störeinflüssen (wechselnde Solarleistung und/oder wechselndem Verbrauch im Haus) möglichst schnell ohne Überschwingen ausgeglichen wird. In der Regelungstechnik verwendet man üblicherweise dazu einen PI-Reglalgorithmus, der die Ansteuerung des Wechselrichters berechnet. Dieser bewertet die Regelabweichung mit einem proportionalen P- und einem integralen I-Anteil. Da es bei meiner Konfiguration zu starken Überschwingern kommt, erscheint mir der P-Anteil zu groß. Die Reaktion des DPR ist bei kleinen Regelabweichungen viel zu stark. Die WR-Leistung schaukelt sich auf und springt im Extremfall zwischen Pmax und 0. Wie kann ich das verhindern?
Kann mir jemand auch bei meinen Fragen 2 und 3 weiterhelfen?

**Comment by Manos1966** [comment, marked as answer]
> Wie kann OpenDTUonBattery den Huawei AC-Lader steuern oder muss die Steuerung extern über den ioBroker erfolgen?

Wiki gelesen?
[https://github.com/helgeerbe/OpenDTU-OnBattery/wiki/Huawei-AC-PSU](https://github.com/helgeerbe/OpenDTU-OnBattery/wiki/Huawei-AC-PSU)

**Reply by dragricola** [reply]
Ja, Wiki aufmerksam studiert und alles so wie beschrieben realisiert. Der Huawei lässt sich auch sowohl im Browser als auch über MQTT gut steuern. Nur ein automatisches Laden, so dass die Netzeinspeisung bei Überschuss aus der PV-Anlage auf Null geregelt wird, stellt sich nicht ein. Im Wiki habe ich zu diesem Thema nichts gefunden. Meine 2. Frage zielt darauf, ob ein solches Verhalten überhaupt vorgesehen ist, oder ob man solch eine Regelung über MQTT in der Homeautomatisation (in meinem Falle ioBroker) selbst realisieren muss.

**Reply by Manos1966** [reply]
Ah so, OK, dann habe ich deine Frage missverstanden
"Wie kann OpenDTUonBattery den Huawei AC-Lader steuern oder muss die Steuerung extern über den ioBroker erfolgen?"

Da es Winter ist und die Solar Produktion eher gering ist, muss ich die Frage stellen:
- Produziert deine Solar Hoymiles ausreichend Strom um einen negativen Wert auf den Stromzaehler zu erreichen?
- Wird dieser "negativer Wert" als Basis benutzt um den Huawei AC Lader zu regeln?
- Wenn das Verhalten wie oben beschrieben, NICHT passiert, wie steuert die OpenDTU den AC-Lader? Welche Werte bekommt der AC-Lader denn, um seiner Produktion anzupassen?

**Reply by dragricola** [reply]
Danke für die Nachfrage! 
> Produziert deine Solar Hoymiles ausreichend Strom um einen negativen Wert auf den Stromzaehler zu erreichen?

Ja, trotz grauem Himmel produziert meine PV mit 6kWpeak immer mal wieder Überschuss. Ich teste natürlich immer nur wenn  Überschuss vorhanden ist und mein Zähler negativen Verbrauch meldet. Meine PV-Anlage hat übrigens 2 Hoymiles HMT2250 und 1 HMT1800 (separat überwacht durch eine normale OpenDTU) und einen Hoymiles MI300 (separat überwacht durch eine Ahoy DTU). Ich verwende keinen Victron, der die Batterie direkt lädt. In openDTUonBattery ist nur der Batteriewechselrichter HM1200 konfiguriert, da die HMT-Wechselrichter nicht zusammen mit dem Huawei AC-Charger kommunizieren können. 
>Wird dieser "negative Wert" als Basis benutzt um den Huawei AC Lader zu regeln?

Der Stromzählertyp: HTTP(S) + JSON (Hichi mit Tasmota) ist in OpenDTUonBattery konfiguriert und das Live-Display meldet unter Leistung / Netz korrekte vorzeichenrichtige Werte.
>Wenn das Verhalten wie oben beschrieben, NICHT passiert, wie steuert die OpenDTU den AC-Lader? Welche Werte bekommt der AC-Lader denn, um seine Produktion anzupassen?

Ja, das genau ist meine Frage! 
Ich ging davon aus, dass durch meine Dynamic Power limiter Konfiguration (siehe oben) openDTUonBattery alle Infos hat, um den AC-Lader zu steuern. Mir scheint der DPL jedoch nur den Hoymiles zu steuern. Habe ich vielleicht bei der DPL Konfiguration etwas falsch verstanden?  Inzwischen habe ich noch Solar Passthrough deaktiviert, da es wohl nur Sinn macht, wenn die Batterie über einen Victron geladen wird. 
Hier liefere ich noch die oben nicht angegebenen weiteren Konfigurationsdaten nach: Akku SoC - Start: 75 %, Akku SoC - Stop 20 %, (Sind diese beiden Werte in meinem Fall nicht irrelevant, da keine Werte von der Batterie kommen?), DC Spannung - Start: 51 V, DC Spannung - Stop: 50 V, DC Spannung - Lastkorrekturfaktor: hier habe ich verschiedene Werte zwischen 0,05 und 0,0013 probiert.  
Es ist auch nicht klar wie "Wechselrichter ist hinter Leistungsmesser" zu konfigurieren ist , da man nicht weiß, ob "hinter" von OpenDTUonBattery- oder von Netz-Seite aus gesehen bedeutet.

**Reply by Manos1966** [reply]
Vorab, "Wechselrichter ist hinter Leistungsmesser" sollte eingeschaltet sein (ich gehe davon aus, wenn dein HM1200 der an die Batterie haengt, Strom einspeist, das wird vom Stromzaehler gemessen. Wie 99,9% bei allen der Fall ist).
Also von Netz-Seite aus gesehen  Netz<-Stromzaehler<- Haus+HM1200 
 
Da ich auch selber in einer Test-Umgebung teste, wo eine dutzend Wechselrichter aktiv sind und 50kWp und dadurch erlebe sehr oft das Problem: Wenn alle Hoymiles (2xHMT2250, HMT1800, HM1200, HM300 gleichzeitig auf den gleichen Stromzähler einspeisen, dann könnte ich mir vorstellen, dass massive Schwankungen tagsüber so lange die Sonne scheint, passieren.

Eine Test-Frage habe ich noch: Nachts, wenn nur der HM1200 von der Batterie Strom einspeist (alle anderen Hoymiles sind aus), funktioniert deine Anlage erwartungsgemäß?

Wenn ich es richtig verstehe, brauchst du die HM1200 nur Nachts. Wie stellst du sicher, dass die HM1200 nur Nachts aktiv wird? (die Strategie zur Batterieentleerung: Leeren nur Nachts ist nur auswaehlbar wenn Solar Passthrough eingeschaltet ist).

**Reply by dragricola** [reply]
> "Wechselrichter ist hinter Leistungsmesser" sollte eingeschaltet sein

OK, erledigt!
>  dass massive Schwankungen tagsüber so lange die Sonne scheint, passieren

ja, das ist mir bewußt. Aber bei PV-Überschuss sollte der AC-Charger nicht anspringen? 
> Nachts, wenn nur der HM1200 von der Batterie Strom einspeist (alle anderen Hoymiles sind aus), funktioniert deine Anlage erwartungsgemäß?

Grundsätzlich ja, jedoch mit der in meiner Antwort an @schlimmchen beschriebenen Instabilität.
> Wie stellst du sicher, dass die HM1200 nur Nachts aktiv wird?

Wenn ich  laden will, trenne ich zum Testen zur Zeit noch manuell den HM1200 vom AC-Netz und schalte den Huawei ein und wenn ich entladen will, ist es umgekehrt.  Später soll das der iobroker über WLAN-Zwischenstecker in Abhängigkeit von der Batteriespannung und vom Vorzeichen der Netzleistung mit passender Hysterese sowohl tags wie nachts automatisch erledigen. 
> Leeren nur Nachts ist nur auswaehlbar wenn Solar Passthrough eingeschaltet ist

Solar Passthrough habe ich deaktiviert. Stimmt es nicht, dass diese Option nur für Systeme sinnvoll ist, die über einen MPTT-Laderegler (z.B. Viktron) die Batterie direkt laden können? Ich habe die Solarwechselrichter und DC-Spannung nur auf dem Dach und führe die PV-Leistung nur als 3-phasige AC-Leistung herunter ins Haus, wo meine Batterie mit Huawei, HM1200 und die DTUs stehen.

**Comment by Manos1966** [comment]
> Die CAN-Schnittstelle der WULILLS 48v 120ah Lifepo4 Battery ist angeblich kompatibel zur Pylontech. Leider konnte bisher noch keine Kommunikation aufgebaut werden. Weiß jemand Rat?

Das hoere ich ab und zu ("Pylontech compatibel" aber....)

Die 100EUR Frage ist aber immer die gleiche: 
Bist du sicher, dass deine CANBus Schnittstelle mit einer Pylontech funktioniert? 
Denn, so lange du nicht 100% prozentig sicher bist, dass deine CANBus Schnittstelle mit einer echten Pylontech funktioniert und mit deiner  WULILLS nicht, kannst du nicht mit Sicherheit sagen, dass es an deiner Batterie liegt.

**Reply by dragricola** [reply]
> Bist du sicher, dass deine CANBus Schnittstelle mit einer Pylontech funktioniert?

Leider nein, da ich keine Pylontech zum Testen habe. Kann ich das vielleicht mit dem Projekt von solaranzeige.de testen? Ich hatte noch keine Zeit, mich damit näher zu beschäftigen. Gibt es hier im Forum Erfahrung damit?

**Reply by Manos1966** [reply]
Na ja, eine Moeglichkeit ist einen Bekannten in deiner Gegend zu finden, der eine Pylontech hat 😉 
Zweite Moeglichkeit, falls du ein Raspberry Pi hast, einen Pylontech Emulator zu bauen (dafuer brauchst du eine extra CANBUS Platine)
wird hie beschrieben: [Pylontech-emulator](https://github.com/helgeerbe/OpenDTU-OnBattery/wiki/Pylontech-battery#pylontech-emulator)

**Reply by dragricola** [reply]
> [Pylontech-emulator](https://github.com/helgeerbe/OpenDTU-OnBattery/wiki/Pylontech-battery#pylontech-emulator)
Danke für den Tipp - werde ich ausprobieren. Die Teile habe ich da!

**Comment by Manos1966** [comment]
> Meine PV-Anlage hat übrigens 
2 Hoymiles HMT2250 und 
1 HMT1800 (separat überwacht durch eine normale OpenDTU) und 
einen Hoymiles MI300 (separat überwacht durch eine Ahoy DTU).

**R E S P E K T!** 👀 

...und gut, dass du es erwaehnt hast.

**Reply by Manos1966** [reply]
@dragricola guck Mal auch hier #515 
bf4ed hat am Ende seine Einstellungen gepostet.

**Reply by dragricola** [reply]
>guck Mal auch hier https://github.com/helgeerbe/OpenDTU-OnBattery/discussions/515

Vielen Dank! Dort hatte ich zuvor geschaut, aber die Beispiele zur Konfiguration sind neu. Ich habe nun die Konfiguration entsprechend angepasst. Vermutlich waren die Start- und Stoppwerte falsch. Leider ist gerade Nebel aufgezogen und die PV-Leistung ist nicht ausreichend zum Testen des AC-Chargers. Ich werde berichten, wenn es wieder PV-Überschuss gegeben hat.

**Reply by dragricola** [reply]
Nachtrag: Da bisher die CAN-Schnittstelle zur Batterie noch nicht funktioniert, habe ich keine Batterie konfiguriert. Somit ist der SOC unbekannt. Basiert in diesem Falle die AC-Charger-Steuerung auf Basis der Ausgangsspannung des AC-Chargers oder dem gleichlautenden Wert der Eingangsspannung des HM1200?

**Reply by Manos1966** [reply]
dem gleichlautenden Wert der Eingangsspannung des HM1200

ABER, ABER, ABER (und dies ist der Grund warum ich persoenlich SOC bzw CANbus nutze), 
da Huawei, Batterie und HM-Eingaenge am gleichen Kabel sind, 
wenn tagsueber die Huawei/Victron die Batterie laedt, 
die Spannung auf die HM-Eingaenge ist die GLEICHE 😉

**Reply by Manos1966** [reply]
Hallo @dragricola  

bf4ed hat uns noch ein paar Tipps gegeben.

Er schaltet seine Batterie-gebundene Hoymiles nur Nachts ein.
Steuerung ueber Shelly 1 Plus PM (feste Uhrzeit)

Damit hat man keine Probleme mit der Produktion der Huawei (Nachts laedt sie die Batterie nicht) und die Eingangsspannung der Hoymiles ist die Spannung der Batterie. 
Kleine Anmerkung: Deswegen ist "Target Power Consumption from Grid" mit Null (0) eingestellt (damit die Huwai nicht anspringt wenn die Hoymiles Strom produziert).

**Reply by dragricola** [reply]
Hallo @Manos1966 ,
vielen Dank für die Tipps. Leider erlaubt das trübe Wetter zur Zeit keinen PV-Überschuß, sodass ich die Steuerung des AC-Cargers  noch nicht testen konnte. Immerhin sind mit den Start-und Stopp-Werten von bf4ed die starken Leistungsschwingungen beim Hoymiles verschwunden. Wenn das Wetter sich gebessert hat, melde ich mich wieder.  Laut Wetterbericht kann das aber noch etwas dauern.

**Reply by dragricola** [reply]
Hallo @Manos1966,
nun ging es doch schneller als gedacht: Die Wolkendecke war nicht mehr so dicht, sodass (nachdem ich den MQTT Topic huawei.cmd.mode auf 3 gesetzt hatte) openDTUonBattery heute automatisch per AC-Charger mit über 900W die Batterie laden konnte! Nochmals vielen Dank für Deine Hilfe! Nun kann ich mich der CAN-Kommunikation mit meiner Batterie widmen.

**Reply by Manos1966** [reply]
@dragricola Schoen zu hoeren, dass so schnell geklappt hat!

In der Tat, es gibt sehr wenige Anwender, die die Huawei nutzen, dementsprechend gibt es wenige Infos.

I war damals so beeindruckt, dass OpenDTUonBattery in der Lage ist eine so komplexe Steuerung zu meistern, dass ich selber einen Huawei 4850 gekauft habe. In der Zwischenzeit sind sechs Monate vergangen und mein Huawei liegt immer noch im Schrank... 🤦‍♂️ (ich bin immer noch mit Victron MPPT & Pylontech am experimentieren)

> Hello @bf4ed and @Manos1966
Thank you very much!
The settings you published helped me to put my very similar system to operation.
One small detail can be added in your documentation: For systems that communicate via MQTT with a home automation system it is important to set the MQTT Topics powerlimiter/cmd/mode to 0, huawei.cmd.mode to 3 and the other Topics under huawei.cmd.limit_... to the battery depending values.

Vielen Dank fuer diese Information.
Kannst du bitte (bei Gelegenheit) hier etwas mehr Informationen darüber posten? 
Da ich mein Huawei nicht installiert habe, brauche etwas mehr Hilfe zum verstehen 🙈 und die Infos in die Wiki zu addieren.

**Reply by dragricola** [reply]
Hallo @Manos1966,
Gerne helfe ich Dir. 
Wenn Du allerdings per Victron die Batterie laden kannst, ist ein zusätzlicher AC-Charger nur sinnvoll, wenn Du noch zusätzliche Panels hast, die nicht am Victron angeschlossen sind. 
Ansonsten funktioniert mit dem noch zusätzlich an den ESP32 anzuschließenden CAN-Adapter die Integration des Huawei problemlos. Auf das Relais kann man notfalls verzichten, wenn man den Slot-detect Anschluss permanent mit dem negativen DC-Anschluss verbindet (mein Huawei startet übrigens auch ohne diese Verbindung) und den Huawei AC-seitig über einen schaltbaren Zwischenstecker anschließt und nur tagsüber oder besser nur bei PV-Überschuss einschaltet. 
Gib Bescheid, wenn Du weitere Informationen benötigst.

**Comment by dragricola** [comment]
Hallo @helgeerbe 
inzwischen habe ich openDTUonBattery mit einem Huawei AC-Charger erfolgreich zum Laufen gebracht. Vielen Dank für die Zurverfügungstellung dieses sehr schönen Projektes mit dessen Hilfe sich die Wirtschaftlichkeit einer PV-Anlage mit Batterie sehr steigern lässt! 
Mir ist jedoch aufgefallen, dass sich bei der Konfiguration für den AC-Charger leider keine Werte für die minimale Lestung kleiner 100 W einstellen lassen. 
Aus welchem Grund besteht diese Einschränkung? 
Im manuellen Betrieb kann der Huawei problemlos auch kleine Leistungen von wenigen Watt einstellen und die Regelung reagiert auf Änderungen in der PV-Produktion oder Laständerungen im Hausnetz sehr schnell. Falls keine gravierenden Gründe dagegensprechen, bitte ich darum, diesen Minimalwert in der nächsten Version herunterzusetzen (auf z.B. 10W) um auch bei geringer Solareinstrahlung, wie es im Winter leider öfter vorkommt, die Batterie öfter und mehr laden zu können.
